### ISSUE TITLE
Siri announcing wrong call format

### ISSUE THREAD

**Issue by wiswedel** [nextcloud/talk-ios#550]
## Steps to reproduce
1. Have "Announce incoming calls" activated in iOS settings
2. Video call comes in
3. Display shows "[Call name] - Nextcloud Talk Video"

### Expected behaviour
Siri audio announcement says: "Nextcloud Talk Video"

### Actual behaviour
Siri audio announcement says: "Nextcloud Talk Audio"

## Device information

**Device:** iPhone 12 mini

**iOS version:** 14.4.2

**Talk version:** 11.1.1

**Comment by SystemKeeper** [comment]
I‘ve noticed this as well... I‘m wondering if this happens because initially it’s always a voice call which gets updated to a video call after a roundtrip to the api...?!

**Comment by wiswedel** [comment]
> initially it’s always a voice call which gets updated to a video call

Good point - meaning that in fact it's wrongfully announced in the displayed text, right?

Maybe the least confusing way would be to pin no format at all and just call it "Nextcloud Talk"?
Ideally Siri would announce the caller/room instead of the app (just like for incoming phone calls where it also doesn't say "Phone")

**Comment by Ivansss** [comment]
@SystemKeeper you are right. We don't know much about the call when receiving the push notification.
Since iOS 13 we need to present the callkit incoming call screen right after receiving a VoIP push notification (if not the OS will kill the app).
After presenting it, we fetch the caller information and the media available in the call (voice-only or video).
I will need to check how we could improve the announcement.

**Comment by Ivansss** [comment]
Thanks for the report @wiswedel ;)

**Comment by SystemKeeper** [comment]
Maybe we can add the flag to the push payload? Although this only helps when the call is initiated as video call.

**Comment by SystemKeeper** [comment]
Just as an update: If the request is fast enough (like on local network) Siri correctly announces "Nextcloud Talk Video", but if the announcement / call starts before our request to the server finished, it always announces "Nextcloud Talk Audio".
Also, sadly, I was unable to find a way to modify what is announced.
